src/wasm/literal.cpp

```diff
@@ -92,6 +107,16 @@ Literal::Literal(const Literal& other) : type(other.type) {
   }
 }
 
+Literal::~Literal() {
+  if (type.isException()) {
+    exn.~unique_ptr();
+  } else if (isGCData()) {
+    gcData.~shared_ptr();
+  } else if (type.isRtt()) {
+    rttSupers.~unique_ptr();
+  }
+}
```

Comment:
It might be good to use a `switch` here so the compiler will complain about unhandled variants in the future.

Reply:
Good idea, I'll add an assert. (Seems shorter than a switch? But maybe I'm not seeing how you would use one.)

Reply:
Right, I forgot that `Literal` uses a `Type` to discriminate between the different variants rather than having its own `LiteralKind` enum to switch on. This looks good, thanks!